### PR TITLE
fix: Pdv 3 user registry proteggere le api di lettura con il namespace derivante da api key

### PR DESCRIPTION
**Pull request by BenitoVisone**
- fix namespace requirement and tests
- update CODEOWNERS file

### CODE REVIEW

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Service
+public class PersonConnectorImpl implements PersonConnector {
```
Comment (rodoviti) [inline connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]:
```suggestion
class PersonConnectorImpl implements PersonConnector {
```

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private final PersonRestClient personRestClient;
+
+    @Autowired
+    PersonConnectorImpl(PersonRestClient personRestClient) {
+        this.personRestClient = personRestClient;
+    }
+
+    @Override
+    @ResponseBody
+    @PutMapping(value = "${rest-client.person.saveNamespacedId.path}", produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
```
Comment (rodoviti) [inline connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]:
```suggestion
```

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    public void saveNamespacedId(String id, String namespace, SavePersonNamespaceDto request) {
+        personRestClient.saveNamespacedId(id, namespace, request);
+    }
+
+    @Override
+    @ResponseBody
+    @PatchMapping(value = "${rest-client.person.save.path}", produces = MediaType.APPLICATION_JSON_VALUE, consumes = MediaType.APPLICATION_JSON_VALUE)
```
Comment (rodoviti) [inline connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]:
```suggestion
```

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    public void save(String id, SavePersonDto request) {
+        personRestClient.save(id, request);
+    }
+
+    @Override
+    @ResponseBody
+    @GetMapping(value = "${rest-client.person.findById.path}", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (rodoviti) [inline connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]:
```suggestion
```

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    @Override
+    public PersonResource findById(String id) {
+        return findById(id, "");
+    }
+
+    @Override
+    @ResponseBody
+    @GetMapping(value = "${rest-client.person.findIdByNamespacedId.path}", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (rodoviti) [inline connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]:
```suggestion
```

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    @Override
+    @ResponseBody
+    @GetMapping(value = "${rest-client.person.findById.path}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public PersonResource findById(String id, String namespace) {
+        return personRestClient.findById(id, namespace);
+    }
+
+    @Override
+    public PersonResource findById(String id) {
+        return findById(id, "");
```
Comment (rodoviti) [inline connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]:
```suggestion
        return findById(id, null);
```

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import it.pagopa.pdv.user_registry.connector.model.SavePersonDto;
+import it.pagopa.pdv.user_registry.connector.model.SavePersonNamespaceDto;
+import it.pagopa.pdv.user_registry.connector.rest.client.PersonRestClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
```
Comment (rodoviti) [inline connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]:
```suggestion
import org.springframework.stereotype.Service;
```

File: core/src/main/java/it/pagopa/pdv/user_registry/core/UserServiceImpl.java [hunk 8]
```diff
@@ -18,6 +18,8 @@ class UserServiceImpl implements UserService {
     private final PersonConnector personConnector;
     private final TokenizerConnector tokenizerConnector;
 
+    public static final String GLOBAL_NAMESPACE = "GLOBAL";
+
```
Comment (rodoviti) [inline core/src/main/java/it/pagopa/pdv/user_registry/core/UserServiceImpl.java]:
```suggestion
```

**Review by rodoviti — CHANGES_REQUESTED** [review]

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java (hunk 1 above)
Comment (BenitoVisone) [reply connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]: without "public" keyword I have problem during tests, view: https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java (hunk 2 above)
Comment (BenitoVisone) [reply connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]: fixed with https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java (hunk 3 above)
Comment (BenitoVisone) [reply connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]: fixed with https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java (hunk 4 above)
Comment (BenitoVisone) [reply connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]: fixed with https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java (hunk 6 above)
Comment (BenitoVisone) [reply connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]: fixed with https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java (hunk 7 above)
Comment (BenitoVisone) [reply connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]: fixed with https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

File: core/src/main/java/it/pagopa/pdv/user_registry/core/UserServiceImpl.java (hunk 8 above)
Comment (BenitoVisone) [reply core/src/main/java/it/pagopa/pdv/user_registry/core/UserServiceImpl.java]: fixed with https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

File: connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java (hunk 5 above)
Comment (BenitoVisone) [reply connector/rest/src/main/java/it/pagopa/pdv/user_registry/connector/rest/PersonConnectorImpl.java]: fixed with https://github.com/pagopa/pdv-ms-user-registry/pull/15/commits/6abc21da5995db09c8d91cf7ca06fd215d48edc9

**Review by rodoviti — APPROVED** [review]

**Pull request merged**